### PR TITLE
Addresses #4175, issues with ft for ElectricLoadCenter:Distribution

### LINKED ISSUE #4175: Openstudio Forward translator failing to translate ElectricLoadCenter:Distribution Inverter name attribute
Issue overview
--------------
Models 
## Current Behavior
OpenStudio models using PVWatts Inverter Objects fail with the following error: (from eplusout.err)

 ** Severe  ** ElectPowerLoadCenter constructor ElectricLoadCenter:Distribution="ELECTRIC LOAD CENTER DISTRIBUTION 1", invalid entry.
   **   ~~~   ** Inverter Name is blank, but buss type requires inverter.
   **  Fatal  ** ElectPowerLoadCenter constructor Preceding errors terminate program

## Expected Behavior
Model should simulate as expected, without EnergyPlus failure 

## Steps to Reproduce

See attached files - .
1) in.osm file with properly specified ElectricLoadCenterDistribution object attributes
2) resulting in.idf file that is produced
 
## Possible Solution
It appears the forward translator is not properly handling the inverter assignment defined in the ElectricLoadCenterDistribution object.

## Details
N/A

### Environment

Some additional details about your environment for this issue (if relevant):
 - OpenStudio v3.1.0 (via OS App v1.1.0)
 - Windows 64 Bit OS

### Context
Not able to properly model ElectricLoadCenterDistribution interaction with PV Watts generation scenarios w/o potentially resorting to a EnergyPlus measure as a work around.

Not able to properly model ElectricLoadCenterDistribution  interaction with PV Watts + Simple Battery Storage either, which is a real drag. #

Proper operation of these features are needed to provide feedback for Grid Interactive Efficient Building Operation scenarios

### PR DESCRIPTION
**Pull request by joseph-robertson**
Pull request overview
---------------------

 - Related to #4175
 - Adds `energyplus/Test/ElectricLoadCenterDistribution_GTest.cpp` test file with:
   - `ForwardTranslator_ElectricLoadCenterDistribution_NoInverter`
   - `ForwardTranslator_ElectricLoadCenterDistribution_NoGenerators` 
- Allows `ElectricLoadCenter:Distribution` objects with no generators/transformers (e.g., `LrgOff_GridStorageDemandLeveling.idf`) by updating `ForwardTranslator.cpp`
- Don't create `ElectricLoadCenter:Distribution` objects in ctors for:
  - `GeneratorFuelCell`
  - `GeneratorMicroTurbine`
  - `GeneratorPhotovoltaic`
  - `GeneratorPVWatts`

Please read [OpenStudio Pull Requests](https://github.com/NREL/OpenStudio/wiki/OpenStudio-Pull-Requests) to better understand the OpenStudio Pull Request protocol.

### Pull Request Author

Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.

 - [ ] Model API Changes / Additions
 - [ ] Any new or modified fields have been implemented in the EnergyPlus ForwardTranslator (and ReverseTranslator as appropriate)
 - [ ] Model API methods are tested (in `src/model/test`)
 - [ ] EnergyPlus ForwardTranslator Tests (in `src/energyplus/Test`)
 - [ ] If a new object or method, added a test in NREL/OpenStudio-resources: Add Link
 - [ ] If needed, added VersionTranslation rules for the objects (`src/osversion/VersionTranslator.cpp`)
 - [ ] Checked behavior in OpenStudioApplication, adjusted policies as needed (`src/openstudio_lib/library/OpenStudioPolicy.xml`)
 - [ ] Verified that C# bindings built fine on Windows, partial classes used as needed, etc.
 - [ ] All new and existing tests passes
 - [ ] If methods have been deprecated, update rest of code to use the new methods

**Labels:**

 - [ ] If change to an IDD file, add the label `IDDChange`
 - [ ] If breaking existing API, add the label `APIChange`
 - [ ] If deemed ready, add label `Pull Request - Ready for CI` so that CI builds your PR

### Review Checklist

This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] Code Style, strip trailing whitespace, etc.
 - [ ] All related changes have been implemented: model changes, model tests, FT changes, FT tests, VersionTranslation, OS App
 - [ ] Labeling is ok
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified

### CODE REVIEW

File: src/energyplus/ForwardTranslator/ForwardTranslateElectricLoadCenterDistribution.cpp [hunk 1]
```diff
@@ -63,26 +63,31 @@ namespace energyplus {
 
     IdfObject idfObject = createRegisterAndNameIdfObject(openstudio::IddObjectType::ElectricLoadCenter_Distribution, modelObject);
 
+    /// Generator Fields
     IdfObject generatorsIdf(openstudio::IddObjectType::ElectricLoadCenter_Generators);
-    generatorsIdf.setName(idfObject.name().get() + " Generators");
-    m_idfObjects.push_back(generatorsIdf);
+    if (!modelObject.generators().empty()) {
+      m_idfObjects.push_back(generatorsIdf);
```
Comment (jmarrec) [inline src/energyplus/ForwardTranslator/ForwardTranslateElectricLoadCenterDistribution.cpp:69]: Probably move the line 67 inside the if block?

**Review by jmarrec — COMMENTED** [review]
I think the changes make sense, the API shouldn't create objects just because it's convenient or for historical reasons. 

Have you tried the regression tests with that PR? Should we maybe create a **quick** one with a valid ELCD with no generators (maybe just a transformer, like in the E+ example files)?

**Comment by jmarrec** [comment]
Added a test at https://github.com/NREL/OpenStudio-resources/pull/123

File: src/energyplus/ForwardTranslator/ForwardTranslateElectricLoadCenterDistribution.cpp (hunk 1 above)
Comment (joseph-robertson) [reply src/energyplus/ForwardTranslator/ForwardTranslateElectricLoadCenterDistribution.cpp]: Along with the other generator stuff?

**Review by jmarrec — APPROVED** [review]

**Comment by jmarrec** [comment]
@joseph-robertson I'm tagging this as API change as this may create crashes for existing workflow (people who depended on the ELCD being instantiated). We should make sure to add this one to the release notes (I added a draft markdown document in #4194, but that isn't merged yet).

**Pull request merged**